### DISCUSSION TITLE
How to reduce latency with foxglove bridge

### DISCUSSION THREAD

**Discussion by mrkbac** [foxglove/community#632] [Q&A]
Currently when connecting with slow internet speed to the bridge, the latency between received messages and the currently published messages in ROS can get quite high.
Is it possible somehow to only keep the newest message in cache and drop all old messages, to make sure the client always gets the newest message?

**Comment by defunctzombie** [comment]
The bridge is mostly designed around having a reliable connection between your robot and Studio. Reliable will take different forms depending on the topics you subscribe to. Due to the architecture of the protocol, the bridge, and the underlying connection logic there is not a great way to do message dropping per-topic at this time. Running over lossy links is something that we will explore in the future.

One option you do have is the `send_buffer_limit` (https://github.com/foxglove/ros-foxglove-bridge#configuration). This is a buffer across all outgoing messages. Once full, new messages are dropped. However, if you make this buffer too small that can have an impact on larger messages you want to send.
